storage-api/src/main/java/com/connexta/transformation/storage/api/status/MetadataInfo.java

```diff
@@ -0,0 +1,102 @@
+/**
+ * Copyright (c) Connexta
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package com.connexta.transformation.storage.api.status;
+
+import com.connexta.transformation.storage.api.ErrorCode;
+import java.io.InputStream;
+import java.util.Optional;
+
+/**
+ * Handles information about a piece of metadata that has been, or will be generated through the
+ * transformation process. The metadata can also be saved or retrieved using this object..
+ */
+public interface MetadataInfo extends TransformationStatus {
+
+  /**
+   * Returns the type of metadata being created (e.g. metacard, irm...).
+   *
+   * @return the type of metadata being created
+   */
+  String getMetadataType();
+
+  /**
+   * Returns an optional {@link InputStream} for retrieving the metadata. If the metadata hasn't
+   * been created yet, or failed to be created, the optional will be empty.
+   *
+   * @return an optional of the input stream for this metadata
+   */
+  Optional<InputStream> getContent();
+
+  /**
+   * Returns an optional containing the content type of the metadata. If the metadata hasn't been
+   * created yet, or failed to be created, the optional will be empty.
+   *
+   * @return an optional containing the content type of the metadata
+   */
+  Optional<String> getContentType();
+
+  /**
+   * Returns an optional containing the length of the metadata. If the metadata hasn't been created
+   * yet, or failed to be created, the optional will be empty.
+   *
+   * @return an optional containing the length of the metadata
+   */
+  Optional<String> getContentLength();
+
+  /**
+   * Used to signify the successful creation of the metadata this MetadataInfo represents, this
+   * method will persist the given content and mark this metadata transformation as complete. The
+   * given {@link InputStream} will be read and closed within this method.
+   *
+   * @param contentType the content type of the metadata
+   * @param contentStream an {@link InputStream} for reading the metadata
+   */
+  void complete(String contentType, InputStream contentStream);
```

Comment:
✏️ Needs to throw StorageException since it's modifying the underlying datastore.